### PR TITLE
Increase vega lite coverage

### PR DESCRIPTION
**Pull request by saulshanabrook**
- [x] Switches back to using vega-embed so we can more easily debug the resulting charts (requires JupyterLab 1.1.0, which is still in beta)
- [x] Fixes https://github.com/Quansight/jupyterlab-omnisci/issues/77 by first calling `extractTransform` before compiling to vega
- [x] Adds support for time based charting
- [x] Fills out mapd charting example https://omnisci.github.io/mapd-charting/example/example1.html

### CODE REVIEW

**Comment by saulshanabrook** [comment]
I am making some good progress on getting a groupby day/month/year example working, but I am realizing that we need to parse the datetime strings into JS datetime  objects, like by using Vega's `toDate` function. It needs to happen before we call `dataflow.ingest` on each row:

https://github.com/Quansight/jupyterlab-omnisci/blob/b8adeaf6f6959b8fc5486cd65f337b8f8a9cd741/src/ibis-transform.ts#L78-L79

To do this, we need to know which of the resulting fields are datetime so we can parse them. @domoritz do  you know of an alternative function to `dataflow.ingest` that would also do this parsing? Or a way that we can know, inside of the transform, what the expected types of the data is?

**Comment by domoritz** [comment]
Vega actually accepts timestamps (numbers) as well. I suspect that you could try to just parse any valid date (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parse) since I can't think of any examples where someone wants the date string to stay a string. 

We don't have a way to tell you what the types of fields are. However, @Saba9 will be working with OmniSci this fall and may be looking into it.

**Comment by saulshanabrook** [comment]
I have the time based charting now working:
Screen Shot 2019-08-10 at 10 12 26 AM


However there is an issue where if I try to click and drag on the timestamp to filter it, I can't unclick to stop selecting. So no matter where I move my mouse, it keeps moving the selection. Also, it seems the UI slows down a lot as I move my mouse in datetime, because it keeps trying to re-query on every movement. To reproduce this, open  this binder link, run all the cells, wait for the charts to render, and try filtering the time chart: 
https://mybinder.org/v2/gh/Quansight/jupyterlab-omnisci/e18b7c6?urlpath=lab/tree/notebooks/Charting%20Example.ipynb

**Comment by saulshanabrook** [comment]
OK this is more workable now. After talking with @domoritz I added a debounce to the signal for the interval selection. In the future this would be made easier by https://github.com/vega/vega-lite/issues/5288. Long term @jheer says, in response to blocking as the user selections:

> In short, no. The browser UI should not block, but Vega event processing will block by simply queueing up the events  for subsequent pipeline evaluations.
> I’d need to think about it more, but I could imagine a future setup that breaks up long-running jobs across two local dataflow executions. But I wouldn’t hold your breath.

File: src/ibis-transform.ts [hunk 1]
```diff
@@ -6,6 +6,29 @@ import * as dataflow from 'vega-dataflow';
 
 import { Transform } from 'vega';
 
+/**
+ * Tries parsing all string values as dates.  Any that cannot be parsed are left alone
+ *
+ * https://github.com/Quansight/jupyterlab-omnisci/pull/85#issuecomment-519656950
+ */
+function parseDates(o: { [key: string]: any }): { [key: string]: any } {
```
Comment (saulshanabrook) [inline src/ibis-transform.ts]: Try seeing what columns are date times on the python side and passing that over.

**Pull request merged**